apps/language_server/lib/language_server/providers/completion.ex

```diff
@@ -608,17 +626,49 @@ defmodule ElixirLS.LanguageServer.Providers.Completion do
       "sortText" => String.pad_leading(to_string(idx), 8, "0"),
       "insertText" => item.insert_text,
       "insertTextFormat" =>
-        if snippets_supported do
+        if {:snippets_supported, true} in options do
           insert_text_format(:snippet)
         else
           insert_text_format(:plain_text)
         end
     }
 
+    # deprecated as of Language Server Protocol Specification - 3.15
+    json =
+      if {:deprecated_supported, true} in options do
+        Map.merge(json, %{
+          "deprecated" => item.tags |> Enum.any?(&(&1 == :deprecated))
+        })
+      else
+        json
+      end
+
+    tag_supported = options |> Keyword.fetch!(:tag_supported)
```

Comment:
Can we change this keyword to be `tags_supported`? (another option would be just `tag_support`, but I prefer the first).